compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9355,20 +9362,32 @@ private BLangBlockStmt getSafeErrorAssignment(Location location, BLangSimpleVarR
                         .anyMatch(retType -> types.isAssignable(errorType, retType)));
 
         String patternFailureCaseVarName = GEN_VAR_PREFIX.value + "t_failure";
-        BLangSimpleVariable errorVar =
-                ASTBuilderUtil.createVariable(location, patternFailureCaseVarName, symTable.errorType,
-                                                createTypeCastExpr(ref, symTable.errorType),
-                                              new BVarSymbol(0, names.fromString(patternFailureCaseVarName),
-                                                             this.env.scope.owner.pkgID, symTable.errorType,
-                                                             this.env.scope.owner, location, VIRTUAL));
+        BVarSymbol errorVarSymbol;
+        if (!isCheckPanicExpr && this.onFailClause != null && invokableSymbol.origin != VIRTUAL) {
+            errorVarSymbol = onFailClauseErrorVariable.computeIfAbsent(this.onFailClause,
+                    k -> new BVarSymbol(0, names.fromString(patternFailureCaseVarName),
+                            this.env.scope.owner.pkgID, symTable.errorType, this.env.scope.owner, null, VIRTUAL));
```

Comment:
Why are we passing the location as null here? earlier we were using the `location`

Reply:
Since the synthetic variables are not exposed to users, no need to add locations for them.